packages/web/hooks/use-swap.tsx

```diff
@@ -1124,45 +1461,81 @@ function useQueryRouterBestQuote(
-      // if failed quotes are being returned
-      // until retry starts returning false.
-      // This causes slow UX even though there's a
-      // quote that the user can use.
-      retry: false,
-
-      // prevent batching so that fast routers can
-      // return requests faster than the slowest router
-      trpc: {
-        context: {
-          skipBatch: true,
-        },
-      },
-    }
-  );
+
+  const quoteResult =
+    quoteType === "out-given-in"
+      ? trpcReact.local.quoteRouter.routeTokenOutGivenIn.useQuery(
+          {
+            tokenInAmount: input.tokenInAmount,
+            tokenInDenom: input.tokenIn?.coinMinimalDenom ?? "",
+            tokenOutDenom: input.tokenOut?.coinMinimalDenom ?? "",
+            forcePoolId: input.forcePoolId,
+          },
+          {
+            enabled: enabled,
+
+            // quotes should not be considered fresh for long, otherwise
+            // the gas simulation will fail due to slippage and the user would see errors
+            staleTime: 5_000,
+            cacheTime: 5_000,
+            refetchInterval: 5_000,
+
+            // Disable retries, as useQueries
+            // will block successfull quotes from being returned
+            // if failed quotes are being returned
+            // until retry starts returning false.
+            // This causes slow UX even though there's a
+            // quote that the user can use.
+            retry: false,
+
+            // prevent batching so that fast routers can
+            // return requests faster than the slowest router
+            trpc: {
+              context: {
+                skipBatch: true,
+              },
+            },
+          }
+        )
+      : trpcReact.local.quoteRouter.routeTokenInGivenOut.useQuery(
+          {
+            tokenOutAmount: input.tokenInAmount ?? "",
+            tokenOutDenom: input.tokenOut?.coinMinimalDenom ?? "",
+            tokenInDenom: input.tokenIn?.coinMinimalDenom ?? "",
+            forcePoolId: input.forcePoolId,
+          },
+          {
+            enabled: enabled,
+
+            // quotes should not be considered fresh for long, otherwise
+            // the gas simulation will fail due to slippage and the user would see errors
+            staleTime: 10_000,
+            cacheTime: 10_000,
+            refetchInterval: 10_000,
+
+            // Disable retries, as useQueries
+            // will block successfull quotes from being returned
+            // if failed quotes are being returned
+            // until retry starts returning false.
+            // This causes slow UX even though there's a
+            // quote that the user can use.
+            retry: false,
+
+            // prevent batching so that fast routers can
+            // return requests faster than the slowest router
+            trpc: {
+              context: {
+                skipBatch: true,
+              },
+            },
+          }
+        );
```

Comment:
It goes against the rule of hooks to conditionally render hooks. Is this working?

Reply:
It is, what's the advised solution here?

Reply:
For reference the call to this hook is duplicated, one for each quote type.

Reply:
If this is the case, I suggest modifying the params themselves using a ternary operator